### ISSUE TITLE
Support FetchRequest v4 / Message Format v2

### ISSUE THREAD

**Issue by dpkp** [dpkp/kafka-python#1119]
> In Kafka 0.11, the structure of the 'MessageSet' and 'Message' were significantly
> changed. Not only were new fields added to support new features like exactly once
> semantics and record headers, but the recursive nature of the previous versions of
> the message format was eliminated in favor of a flat structure. A 'MessageSet' is
> now called a 'RecordBatch', which contains one or more 'Records' (and not 'Messages').
> When compression is enabled, the RecordBatch header remains uncompressed,
> but the Records are compressed together. Further, multiple fields in the 'Record'
> are varint encoded, which leads to significant space savings for larger batches. 

See https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-Messagesets

**Comment by tvoinarovskyi** [comment]
fml, started poking with the new format. If we start using it the introduced CRC32C will be a problem probably.
See https://issues.apache.org/jira/browse/KAFKA-1449 for more info.
tl;dr The v2 Message format uses a different CRC32 implementation.

**Comment by tvoinarovskyi** [comment]
A bit of status report here. I finished a pure python prototype on top of aiokafka, currently checking compatibility and tests. On next week plan to port it to kafka-python. Thou I think this will require a C extension to be relatively fast (pure python Crc32C and varint coder can be made much faster), did not measure yet thou.

**Comment by tvoinarovskyi** [comment]
I think this can be closed already
